### PR TITLE
ICU-20200 Treat select compiler warnings as errors in the ICU4C library code (with MSVC)

### PR DESCRIPTION
**Pull request by jefgen**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20200
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/allinone/Build.Windows.PlatformToolset.props [hunk 1]
```diff
@@ -18,9 +18,4 @@
   <PropertyGroup Label="PlatformToolset">
     <PlatformToolset>$(DefaultPlatformToolset)</PlatformToolset>
   </PropertyGroup>
-  <PropertyGroup>
-    <!-- Disable MSBuild warning about Linker OutputFile. -->
-    <!-- Ex: MSBuild complains that the common project creates "icuuc62.dll" rather than "common.dll". However, this is intentional. -->
-    <MSBuildWarningsAsMessages>MSB8012</MSBuildWarningsAsMessages>
-  </PropertyGroup>
```
Comment (daniel-ju) [inline icu4c/source/allinone/Build.Windows.PlatformToolset.props]: I think deleting the warning suppression in this file brought back some of the MSB8012 warnings.

Comment (jefgen) [reply icu4c/source/allinone/Build.Windows.PlatformToolset.props]: Oh good point. I moved it into the "library" file, but not all the projects include the "library" props file. I think we'll need to keep it in both files for now.

Thanks for pointing this out!

Comment (jefgen) [reply icu4c/source/allinone/Build.Windows.PlatformToolset.props]: Please take a look at PR #373. 
(Thanks again!)

Comment (daniel-ju) [reply icu4c/source/allinone/Build.Windows.PlatformToolset.props]: Thanks for making the fix! I approved PR #373. :)

**Pull request merged**